main.tf

```diff
@@ -26,6 +26,8 @@ locals {
   ) : null
   # tflint-ignore: terraform_unused_declarations
   validate_metrics = var.plan != "enterprise-3nodes-2tb" && length(var.metrics) > 0 ? tobool("metrics are only supported for enterprise plan") : true
+  # tflint-ignore: terraform_unused_declarations
+  validate_schema_global_rule = var.plan != "enterprise-3nodes-2tb" && var.schema_global_rule != "" ? tobool("schema global rule is only supported for enterprise plan") : true
```

Comment:
We already have valdiation in place for other features to check for Enterprise plan and this can also be added here-

https://github.com/terraform-ibm-modules/terraform-ibm-event-streams/blob/01ae6f3887d9b89511c7cc62a2a20318c53e0888/main.tf#L8

I have update the validation for metrics also here in this [PR](https://github.com/terraform-ibm-modules/terraform-ibm-event-streams/pull/327/files#diff-dc46acf24afd63ef8c556b77c126ccc6e578bc87e3aa09a931f33d9bf2532fbb).

Reply:
There is no harm in having separate regex check here - infct its better ad the error message can be more explicit to consumers

Reply:
ok, lets leave it then and will have seperate for each feature.